ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+                    compute_v3_user_localized_key = get(snmp_settings, 'compute_v3_user_localized_key')
+                    if (
+                            version == 'v3' and
+                            compute_local_engineid is True and
+                            compute_v3_user_localized_key is True
+                    ):
+                        user_dict.update({'localized': local_engine_id})
+
+                    auth = get(user, 'auth')
+                    auth_passphrase = get(user, 'auth_passphrase')
+                    if (
+                            version == 'v3' and
+                            auth is not None and
+                            auth_passphrase is not None
+                    ):
+                        user_dict.update({'auth': auth})
+                        hash_filter = {}
+                        if (
+                                compute_local_engineid is True and
+                                compute_v3_user_localized_key is True
+                        ):
+                            hash_filter.update({"passphrase": auth_passphrase, "auth": auth,
+                                                "engine_id": local_engine_id})
+                            user_dict.update({'auth_passphrase': self._hash_passphrase(hash_filter)})
+                        else:
+                            user_dict.update({'auth_passphrase': auth_passphrase})
+
+                        priv = get(user, 'priv')
+                        priv_passphrase = get(user, 'priv_passphrase')
+                        if (
+                            priv is not None and
+                            priv_passphrase is not None
+                        ):
+                            user_dict.update({'priv': priv})
+                            if (
+                                    compute_local_engineid is True and
+                                    compute_v3_user_localized_key is True
+                            ):
+                                hash_filter.update({"passphrase": priv_passphrase, "priv": priv})
+                                user_dict.update({'priv_passphrase': self._hash_passphrase(hash_filter)})
+                            else:
+                                user_dict.update({'priv_passphrase': priv_passphrase})
+                    snmp_server['users'].append(user_dict)
+            return snmp_server
+        return None
+
+    @cached_property
+    def spanning_tree(self):
+        spanning_tree_root_super = get(self._switch_data, 'spanning_tree_root_super')
+        spanning_tree_mode = get(self._switch_data, 'spanning_tree_mode')
+        if spanning_tree_root_super is True or spanning_tree_mode is not None:
+            spanning_tree = {}
+            if spanning_tree_root_super is True:
+                spanning_tree.update({'root_super': True})
+            if spanning_tree_mode is not None:
+                spanning_tree.update({'mode': spanning_tree_mode})
```

Comment:
use a variable to assign `get(self._switch_data, 'spanning_tree_priority', ;32768')` and use it in the code for readbility

Reply:
this goes in general. We need more helper variables.